tests/MongoDB.Driver.Tests/FindFluentTests.cs

```diff
@@ -74,49 +74,162 @@ public class FindFluentTests
         [Theory]
         [ParameterAttributeData]
         public void Count_should_call_collection_Count(
+            [Values(false, true)] bool usingSession,
```

Comment:
This test wasn't testing with sessions and hence a bug was going undetected.

This test has been refactored to test with sessions also.